x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -36,18 +36,61 @@
         {"F", "S"},
         {"z", "X"}
     };
+
+    private static final String[][] JAVA_TIME_FORMAT_REPLACEMENTS_FOR_MYSQL = {
```

Comment:
It's great that you've added some "real world" tests with the format specifiers, but I think it'd be great if you also added a simple one for each one of them too.
For instance `c`, `e` and `H` aren't tested at all anywhere (I didn't go further, there might be some others) and `e` and `D` are wrong. (The latter is revealed by the `DateTimeFormatProcessorTests#testFormatting` test, btw, [running the tests](DateTimeFormatProcessorTests.testFormatting) should help; use `-p x-pack/plugin/sql` to limit the scope of the tests run).